src/Symfony/Component/DependencyInjection/Annotation/Service.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Annotation;
+
+/**
+ * @Annotation
+ * @Target({"CLASS"})
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class Service
+{
+    private $shared;
+
+    private $public;
+
+    private $synthetic;
+
+    private $abstract;
+
+    private $lazy;
+
+    public function __construct(array $data)
```

Comment:
why not removing the constructor and using public properties, letting the annotation parser validate properties itself ? This is the recommended usage of doctrine/annotations when annnotation classes don't need to be reused for other purposes (which is why we don't do in the validator component: constraints themselves are used as annotations, but they are not only annotations)